src/views/Timesheets/components/ShiftItem/index.js

```diff
@@ -0,0 +1,139 @@
+import React, { Component } from 'react';
+import {
+  Typography,
+  Grid,
+  Button,
+  IconButton,
+  Dialog,
+  DialogContent,
+  DialogTitle,
+} from '@material-ui/core';
+import { gordonColors } from '../../../../theme';
+import DeleteForeverOutlinedIcon from '@material-ui/icons/DeleteForeverOutlined';
+
+export default class ShiftItem extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showDeleteConfirmation: false,
+    };
+  }
+
+  handleSubmitButtonClick = () => {
+    this.setState({ showDeleteConfirmation: true });
+  }
+
+  onClose = () => {
+    this.setState({ showDeleteConfirmation: false });
+  }
+
+  render() {
+    const shift = this.props.value;
+    const { deleteShift } = this.props;
+    const {
+      ID,
+      EML_DESCRIPTION,
+      SHIFT_START_DATETIME,
+      SHIFT_END_DATETIME,
+      HOURLY_RATE,
+      HOURS_WORKED,
+      STATUS,
+    } = shift;
+
+    const monthIn = SHIFT_START_DATETIME.substring(5, 7);
+    const dateIn = SHIFT_START_DATETIME.substring(8, 10);
+    const timeIn = SHIFT_START_DATETIME.substring(11, 16);
+
+    const monthOut = SHIFT_END_DATETIME.substring(5, 7);
+    const dateOut = SHIFT_END_DATETIME.substring(8, 10);
+    const timeOut = SHIFT_END_DATETIME.substring(11, 16);
+
+    const dateTimeIn = monthIn + '/' + dateIn + "\n" + timeIn;
+    const dateTimeOut = monthOut + '/' + dateOut + "\n" + timeOut;
+
+    let confirmationBox = (
+      <Grid container>
+        <Grid item>
+          <Dialog
+            open={this.state.showDeleteConfirmation}
+            keepMounted
+            align="center"
+            onBackdropClick={this.onClose}
+          >
+            <DialogTitle>Are you sure you want to delete this shift?</DialogTitle>
+            <DialogContent>
+              <Grid container>
+                <Grid item xs={6} sm={6} md={6} lg={6}>
+                  <Button color="primary" onClick={this.onClose} variant="contained">
+                    No
+                  </Button>
+                </Grid>
+                <Grid item xs={6} sm={6} md={6} lg={6}>
+                  <Button
+                    variant="contained"
+                    onClick={() => {
+                      deleteShift(ID, EML_DESCRIPTION);
+                      this.onClose();
+                    }}
+                    style={styles.redButton}>
+                    Yes
+                  </Button>
+                </Grid>
+              </Grid>
+            </DialogContent>
+          </Dialog>
+        </Grid>
+      </Grid>
+    );
+
+    let shiftItemIcon = (STATUS !== 'Submitted' && STATUS !== 'Approved') ? (
```

Comment:
Would it be more readable to ask whether the status is ‘Saved’ or ‘Rejected’ instead of asking what it is not?

Reply:
Yeah, I had mentioned it while we were working on it and decided since it was working as is, might as well keep it. But changing it would make it more readable